### ISSUE TITLE
Input is required when validator is present, despite require: false

### ISSUE THREAD

**Issue by japerry911** [kestra-io/kestra#4062]
### Describe the issue

Input is required when validator is present, despite require: false

- *This occurs in Kestra Cloud

```yaml
  - id: start_date
    type: STRING
    required: false
    validator: ^\d{4}-\d{2}-\d{2}$
    description: The start date to fetch data from Google Ads.
```

Error Message from UI:

<img width="804" alt="Screenshot 2024-06-17 at 7 04 15 AM" src="https://github.com/kestra-io/kestra/assets/21957238/c05d8d4f-425b-4ee1-a9cf-d889b814634c">

### Environment

- Kestra Version: 0.17.3
- Operating System (OS/Docker/Kubernetes): Cloud
- Java Version (if you don't run kestra in Docker): n/a

**Comment by loicmathieu** [comment]
Just tested it with the following flow and it works:

```yaml
id: input-required
namespace: company.team

inputs:
  - id: start_date
    type: STRING
    required: false
    validator: ^\d{4}-\d{2}-\d{2}$
    description: The start date to fetch data from Google Ads.

tasks:
  - id: hello
    type: io.kestra.plugin.core.log.Log
    message: Hello World! 🚀
```

The issue should have been fixed in 0.18 by another change.
Can you please test again and confirm it's fixed?

**Comment by japerry911** [comment]
Works great, thank you 🙏

**Comment by loicmathieu** [comment]
Fixed in 0.18
